packages/headless/src/composables/menuButton/createMenuButton.ts

```diff
@@ -0,0 +1,72 @@
+import { computed, ref } from "vue";
+import { createBuilder } from "../../utils/builder";
+import { createId } from "../../utils/id";
+
+/**
+ * Purpose of the onSelect function is to set selected option (page) as active
+ */
+export type CreateMenuButtonOptions = { onSelect: (value: string) => void };
+
+export const createMenuButton = createBuilder((options: CreateMenuButtonOptions) => {
+  const menuId = createId("menu");
+  const buttonId = createId("menu-button");
+  const isExpanded = ref<boolean>(false);
+  let timeout: ReturnType<typeof setTimeout> | undefined;
+
+  /**
+   * Debounced hodden state that will only be toggled after a given timeout.
+   */
+  const debouncedHidden = computed({
+    get: () => isExpanded.value,
+    set: (newValue) => {
+      clearTimeout(timeout);
+      timeout = setTimeout(() => {
+        isExpanded.value = newValue;
+      }, 400);
+    },
+  });
+
+  const hoverEvents = computed(() => {
+    return {
+      onMouseover: () => (debouncedHidden.value = true),
+      onMouseout: () => (debouncedHidden.value = false),
+      onFocusin: () => (isExpanded.value = true),
+      onFocusout: () => (isExpanded.value = false),
+    };
+  });
+
+  return {
+    state: { isExpanded },
+    elements: {
+      button: computed(
+        () =>
+          ({
+            "aria-controls": menuId,
+            "aria-expanded": isExpanded.value,
+            "aria-haspopup": true,
+            id: buttonId,
+            ...hoverEvents.value,
+          }) as const,
+      ),
+      listItem: {
+        role: "none",
+      },
+      flyout: {
```

Comment:
Isn't the flyout the same as the menu? So we could get rid of the flyout and bind it to the menu directly

Reply:
We are binding the menu only to the <ul> element, and in the userMenu component where we have header and footer the hovering effects doesn't work. This is why I added the flyout.